packages/xo-server/src/index.js

```diff
@@ -283,12 +284,17 @@ async function registerPlugin(pluginPath, pluginName) {
   } = plugin
   let instance
 
-  const handleFactory = factory =>
+  const config = this._config
+  const handleFactory = async factory =>
     typeof factory === 'function'
       ? factory({
+          config: merge(
+            await readFile(`${pluginPath}/config.toml`).then(TOML.parse, noop),
```

Comment:
This solution is interesting, I did not think about it.

But I feel like this could be handled by the plugin itself.

Reply:
I think it's an implementation choice. There have multiple solutions:

- **Solution 1**: do it generically for all the plugins
- **Solution 2**: only do it for this plugin but the customized configuration will be generic
- **Solution 3**: all the implementation will be specific to the `audit` plugin and `xo-server` will only pass the configuration to it

Reply:
Here's my proposal:
- `xo-server` generically pass `config?.plugins?[pluginName]` to plugins
- `xo-server-audit` has its own default config statically in its code and merges it with the one passed by `xo-server`